Merge the development deltas into a single perldelta

book opened the pull request:
This covers almost all cases. The goal is that the release manager will only have the editing work left, and not have to spend time copy/pasting sections from the individual deltas into the final one.

Solves #23365.

* This set of changes does not require a perldelta entry.

tonycoz commented on Porting/merge-deltas.pl:
```diff
@@ -0,0 +1,329 @@
+my $master = tree_for($template_file);
+
+# loop over all the development deltas
+for my $delta ( map tree_for($_), sort glob "pod/perl$tag_devel*delta.pod" ) {
```
A simple sort will put `perl54310delta.pod` before `perl5432delta.pod`.

Grinnz replied:
Another argument for a new perldelta naming convention for perl ~42~ 44

book committed:
add a script that merges all devel deltas into the final one

For now, the script simply copies the content of each relevant section
into the same one in the master document (pod/perldelta.pod).

It will die when encountering an unexpected =head1 header.

book committed:
process the deltas as utf8 files

book committed:
remove unmodified templated sections from the master

book commented:
And this new version removes unmodified template sections from the final document.

book committed:
add a test script for Porting/merge-deltas.pl

book committed:
update MANIFEST

book committed:
handle running the test from t/ instead of the root of the repository

book committed:
use t/test.pl for testing

ericherman committed:
test merge-deltas.pl dies on unexpected =head1

Co-authored-by: Philippe Bruhat (BooK) <book@cpan.org>

ericherman committed:
Document BLEAD-FINAL perldelta process

Co-authored-by: Philippe Bruhat (BooK) <book@cpan.org>

thibaultduponchelle commented on Porting/merge-deltas.pl:
```diff
@@ -0,0 +1,243 @@
+        next
+          unless ref( $tree->[$pos] ) eq 'ARRAY';
+        next unless $tree->[$pos][0] =~ /\Ahead([1-4])\z/;
+	next if $1 > $level;
```
Back indent expected?

book replied:
TAB characters got in.

ericherman replied:
D'oh. That was pro'ly my fault. Sorry.

thibaultduponchelle commented on Porting/release_managers_guide.pod:
```diff
@@ -478,8 +478,17 @@ time can save you from having to work out the details during the actual
+
+    ./perl -Ilib Porting/merge-deltas.pl
+
+The resulting F<pod/perldelta.pod> will need to be edited.  In particular,
```
`edited.  In` 
Extra space?

book replied:
Yes, double space before period is one of those weird English typography rules that Perl documentation follows.

thibaultduponchelle commented on Porting/release_managers_guide.pod:
```diff
@@ -478,8 +478,17 @@ time can save you from having to work out the details during the actual
 Read F<Porting/how_to_write_a_perldelta.pod>, and try to make sure that
-every section it lists is, if necessary, populated and complete. Copy
-edit the whole document.
+every section it lists is, if necessary, populated and complete.
```
I think I would have put this section out of a this paragraph and in its own, guarded by a `=for checklist`

But it's not a blocker to merge this change :)

thibaultduponchelle commented on t/porting/merge-deltas.t:
I would not use an actual existing CVE number

book replied:
Ah ah, I picked a something that looked invented. Will change the text.

thibaultduponchelle reviewed (approved):
Thank you for that, it's resolving a release management toil
